### PR TITLE
Move THNN Reduction to ATen/core.

### PR DESCRIPTION
**Pull request by gchanan**
This is part of moving the (base) Type to ATen/core; Some Type methods have default argument of type THNN Reduction.

### CODE REVIEW

**Comment by gchanan** [comment]
CC @li-roy.

File: aten/src/ATen/core/Reduction.h [hunk 1]
```diff
@@ -1,5 +1,4 @@
-#ifndef REDUCE_H
-#define REDUCE_H
+#pragma once
 
 namespace Reduction {
```
Comment (ezyang) [inline aten/src/ATen/core/Reduction.h]: Not your fault... but what a weird namespace...

**Review by ezyang — APPROVED** [review]

File: aten/src/ATen/core/Reduction.h (hunk 1 above)
Comment (li-roy) [reply aten/src/ATen/core/Reduction.h]: Just curious, what's the appropriate way to do this? Put it in detail? I wasn't able to use a scoped enum here, but I still wanted them together in one namespace.

**Review by li-roy — APPROVED** [review]

File: aten/src/ATen/core/Reduction.h (hunk 1 above)
Comment (gchanan) [reply aten/src/ATen/core/Reduction.h]: Could you not used a scoped enum because it is C++ or is there another reason?

**Pull request closed without merging**